File: scripts/local-diag.mjs
import Database from 'better-sqlite3'

const DB_PATH = 'C:\\Users\\HP\\AppData\\Roaming\\restaurant-app\\data\\dev.db'
const db = new Database(DB_PATH, { readonly: true })

// What restaurant is configured locally?
const restaurant = db.prepare('SELECT id, name, syncRestaurantId FROM Restaurant LIMIT 5').all()
console.log('\nLOCAL Restaurant:', JSON.stringify(restaurant, null, 2))

// Who is the logged-in user?
const users = db.prepare('SELECT id, email, role, restaurantId, branchId FROM User LIMIT 10').all()
console.log('\nLOCAL Users:', JSON.stringify(users, null, 2))

// Dishes count
const dishCount = db.prepare('SELECT COUNT(*) as count FROM Dish').get()
console.log('\nLOCAL Dish count:', dishCount)

// Inventory count
const invCount = db.prepare('SELECT COUNT(*) as count FROM InventoryItem').get()
console.log('\nLOCAL InventoryItem count:', invCount)

// Transactions (unsynced)
const txUnsynced = db.prepare('SELECT COUNT(*) as count FROM Transaction WHERE synced = 0').get()
const txTotal = db.prepare('SELECT COUNT(*) as count FROM Transaction').get()
console.log('\nLOCAL Transactions - total:', txTotal, '| unsynced:', txUnsynced)

// SyncOutbox
const outbox = db.prepare('SELECT entityType, COUNT(*) as count FROM SyncOutbox GROUP BY entityType').all()
console.log('\nLOCAL SyncOutbox by type:', JSON.stringify(outbox, null, 2))

// Branches
const branches = db.prepare('SELECT id, name, isMain, restaurantId FROM RestaurantBranch LIMIT 10').all()
console.log('\nLOCAL Branches:', JSON.stringify(branches, null, 2))

db.close()

File: scripts/local-diag.py
import sqlite3, json

db = sqlite3.connect(r'C:\Users\HP\AppData\Roaming\restaurant-app\data\dev.db')
c = db.cursor()

# Get restaurant sync info
c.execute("SELECT syncRestaurantId, syncToken, name FROM restaurants LIMIT 3")
rows = c.fetchall()
print("Restaurant sync info:", rows)

# Get tables in the DB
c.execute("SELECT name FROM sqlite_master WHERE type='table' ORDER BY name")
tables = [r[0] for r in c.fetchall()]
print("Tables:", tables)

# Count recent transactions
if 'transactions' in tables:
    c.execute("SELECT COUNT(*) FROM transactions")
    print("Transaction count:", c.fetchone()[0])
    c.execute("SELECT id, amount, type, description, createdAt, synced FROM transactions ORDER BY createdAt DESC LIMIT 5")
    txs = c.fetchall()
    print("Recent transactions:", txs)

# Check sync outbox
if 'sync_outbox' in tables:
    c.execute("SELECT COUNT(*) FROM sync_outbox WHERE syncedAt IS NULL")
    print("Pending outbox items:", c.fetchone()[0])
    c.execute("SELECT entityType, operation, attempts, lastError FROM sync_outbox WHERE syncedAt IS NULL LIMIT 10")
    print("Pending outbox sample:", c.fetchall())

db.close()
